sdk/src/signer/mod.rs

```diff
@@ -103,6 +109,25 @@ pub fn unique_signers(signers: Vec<&dyn Signer>) -> Vec<&dyn Signer> {
     signers.into_iter().unique_by(|s| s.pubkey()).collect()
 }
 
+/// The `EncodableKey` trait defines the interface by which cryptographic
+/// keys/keypairs are read, written, and derived from sources.
+pub trait EncodableKey: Sized {
```

Comment:
can you please squash all changes related to the introduction of this trait and its implementation on `Keypair` down to isolated commits and factor them out to a separate "foundational" PR?